packages/nextcloud/lib/src/version_supported.dart

```diff
@@ -1,33 +1,68 @@
 part of '../nextcloud.dart';
 
+class _SupportedVersion<T> {
+  const _SupportedVersion(
+    this.app,
+    this.version,
+  );
+
+  final String app;
+  final T version;
+}
+
+/// Version of core/Server supported
+const coreSupportedVersion = _SupportedVersion('core', 26);
+
+/// API version of the news app supported
+const newsSupportedVersion = _SupportedVersion('news', 'v1-3');
+
+/// API version of the notes app supported
+const notesSupportedVersion = _SupportedVersion('notes', 1);
+
+/// All supported versions
+const supportedVersions = [
+  coreSupportedVersion,
+  newsSupportedVersion,
+  notesSupportedVersion,
+];
+
 // ignore: public_member_api_docs
 extension CoreVersionSupported on NextcloudCoreClient {
   /// Checks if the app on the server is supported by the client
-  Future<bool> isSupported([final NextcloudCoreServerCapabilities_Ocs_Data? capabilities]) async =>
-      (capabilities ?? (await getCapabilities()).ocs.data).version.major == 26;
+  Future<(bool, _SupportedVersion<int>)> isSupported([
+    final NextcloudCoreServerCapabilities_Ocs_Data? capabilities,
+  ]) async =>
+      (
+        (capabilities ?? (await getCapabilities()).ocs.data).version.major == coreSupportedVersion.version,
+        coreSupportedVersion,
+      );
 }
 
 // ignore: public_member_api_docs
 extension NewsVersionSupported on NextcloudNewsClient {
   /// Checks if the app on the server is supported by the client
-  Future<bool> isSupported() async {
+  Future<(bool, _SupportedVersion<String>)> isSupported() async {
```

Comment:
Somehow I dislike this interface.

Tuples (or however dart calls them) got introduced so we don't need to create such dummy objects and we still have one here.

Also doesn't the caller already know the app they are testing for?
I'd suggest `(isSupported, version)` or if we really need the app name `(isSupported, version, appName)`

Reply:
Hm yeah you're right, the app doesn't really add any info